crates/sui-types/src/object.rs

```diff
@@ -376,7 +376,7 @@ impl MoveObject {
                 *balances.entry(type_tag).or_insert(0) += balance;
             }
         } else {
-            let layout = layout_resolver.get_annotated_layout(self)?;
+            let layout = layout_resolver.get_annotated_layout(&self.type_().clone().into())?;
```

Comment:
nit (optional): I would maybe consider creating a helper function for this on the trait, with a default implementation?